From 310f7206ce149c1a186ed59e473ce5a8d50637af Mon Sep 17 00:00:00 2001
From: Marco De Lucia <delucia@gfz-potsdam.de>
Date: Wed, 7 Jan 2026 11:56:11 +0100
Subject: [PATCH] Fix compatibility with HDF5 2.0.0

    Replace removed H5FD_*_init() functions with predefined constants.
    HDF5 2.0.0 removed internal file driver initialization functions
    in favor of using predefined constants directly.
---
 src/1_12_0/Wrapper_auto_H5FDfamily.c | 2 +-
 src/1_12_0/Wrapper_auto_H5FDlog.c    | 2 +-
 src/1_12_0/Wrapper_auto_H5FDsec2.c   | 2 +-
 src/1_12_0/Wrapper_auto_H5FDstdio.c  | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/1_12_0/Wrapper_auto_H5FDfamily.c b/src/1_12_0/Wrapper_auto_H5FDfamily.c
index 597ca47..c364371 100644
--- a/src/1_12_0/Wrapper_auto_H5FDfamily.c
+++ b/src/1_12_0/Wrapper_auto_H5FDfamily.c
@@ -20,7 +20,7 @@
 /* H5_DLL hid_t H5FD_family_init(void); */
 SEXP R_H5FD_family_init(void){
   int vars_protected=0;
-  hid_t return_val = H5FD_family_init();
+  hid_t return_val = H5FD_FAMILY;
   SEXP R_return_val= R_NilValue;
   R_return_val = PROTECT(ScalarInteger64_or_int(return_val));
   vars_protected++;
diff --git a/src/1_12_0/Wrapper_auto_H5FDlog.c b/src/1_12_0/Wrapper_auto_H5FDlog.c
index 364c10a..0b62be2 100644
--- a/src/1_12_0/Wrapper_auto_H5FDlog.c
+++ b/src/1_12_0/Wrapper_auto_H5FDlog.c
@@ -20,7 +20,7 @@
 /* H5_DLL hid_t H5FD_log_init(void); */
 SEXP R_H5FD_log_init(void){
   int vars_protected=0;
-  hid_t return_val = H5FD_log_init();
+  hid_t return_val = H5FD_LOG;
   SEXP R_return_val= R_NilValue;
   R_return_val = PROTECT(ScalarInteger64_or_int(return_val));
   vars_protected++;
diff --git a/src/1_12_0/Wrapper_auto_H5FDsec2.c b/src/1_12_0/Wrapper_auto_H5FDsec2.c
index 5e4cb40..804c414 100644
--- a/src/1_12_0/Wrapper_auto_H5FDsec2.c
+++ b/src/1_12_0/Wrapper_auto_H5FDsec2.c
@@ -20,7 +20,7 @@
 /* H5_DLL hid_t H5FD_sec2_init(void); */
 SEXP R_H5FD_sec2_init(void){
   int vars_protected=0;
-  hid_t return_val = H5FD_sec2_init();
+  hid_t return_val = H5FD_SEC2;
   SEXP R_return_val= R_NilValue;
   R_return_val = PROTECT(ScalarInteger64_or_int(return_val));
   vars_protected++;
diff --git a/src/1_12_0/Wrapper_auto_H5FDstdio.c b/src/1_12_0/Wrapper_auto_H5FDstdio.c
index 5d6831a..ce710a9 100644
--- a/src/1_12_0/Wrapper_auto_H5FDstdio.c
+++ b/src/1_12_0/Wrapper_auto_H5FDstdio.c
@@ -20,7 +20,7 @@
 /* H5_DLL hid_t H5FD_stdio_init(void); */
 SEXP R_H5FD_stdio_init(void){
   int vars_protected=0;
-  hid_t return_val = H5FD_stdio_init();
+  hid_t return_val = H5FD_STDIO;
   SEXP R_return_val= R_NilValue;
   R_return_val = PROTECT(ScalarInteger64_or_int(return_val));
   vars_protected++;
